src/remarx/sentence/corpus/tei_input.py

```diff
@@ -83,15 +128,43 @@ def text_contents(self) -> Generator[str]:
                 # OR if text comes immediately after (is_tail) and is whitespace only
                 continue
 
-            # consolidate whitespace if it includes a newline
-            # (i.e., space between indented tags in the XML)
-            yield re.sub(r"\s*\n\s*", "\n", text)
+            body_text_parts.append(text)
+
+        # consolidate whitespace once after joining all parts
+        # (i.e., space between indented tags in the XML)
+        return re.sub(r"\s*\n\s*", "\n", "".join(body_text_parts)).strip()
+
+    def get_individual_footnotes(self) -> Generator[str]:
+        """
+        Get individual footnote content as a generator.
+        Yields each footnote's text content individually as a separate string element.
+        Each yielded element corresponds to one complete footnote from the page.
+        """
+        for footnote in self.get_page_footnotes():
+            footnote_text = str(footnote).strip()
+            # consolidate whitespace for footnotes
+            footnote_text = re.sub(r"\s*\n\s*", "\n", footnote_text)
+            yield footnote_text
+
+    def get_footnote_text(self) -> str:
+        """
+        Get all footnote content as a single string, with footnotes separated by double newlines.
+        """
+        footnotes = self.get_page_footnotes()
+        if not footnotes:
+            return ""
+        # consolidate whitespace once after joining all footnotes
+        return re.sub(
+            r"\s*\n\s*",
+            "\n",
+            "\n\n".join(str(footnote).strip() for footnote in footnotes),
```

Comment:
Please update to use `get_individual_footnotes` method. No need to bail out early, if the list is empty join will return an empty string.